src/azure/azureController.ts

```diff
@@ -93,15 +98,37 @@ export class AzureController {
 		this.azureMessageDisplayer = new AzureMessageDisplayer();
 	}
 
+	private async promptForTenantChoice(account: AzureAccount, profile: ConnectionProfile): Promise<void> {
+		let tenantChoices: INameValueChoice[] = account.properties.tenants?.map(t => ({ name: t.displayName, value: t }));
+		if (tenantChoices.length === 1) {
+			profile.tenantId = tenantChoices[0].value.id;
+			return;
+		}
+		let tenantQuestion: IQuestion = {
+			type: QuestionTypes.expand,
+			name: LocalizedConstants.tenant,
+			message: LocalizedConstants.azureChooseTenant,
+			choices: tenantChoices,
+			shouldPrompt: (answers) => profile.isAzureActiveDirectory() && tenantChoices.length > 1,
```

Comment:
nit: don't think you need `tenantChoices.length > 1` since you return in the if above, but it doesn't hurt to make preconditions clear either.

Reply:
I'll keep it here to make it clear. Plus if tenantChoices.length becomes zero, this question should also not prompt.